src/modules/services/submodule/buySiriusCoins/views/ViewServicesStableCoinsCheckStatus.vue

```diff
@@ -0,0 +1,201 @@
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { computed, ref, watch, shallowRef } from "vue";
+import { useRoute } from "vue-router";
+import { networkState } from "@/state/networkState";
+import { ChainSwapConfig } from "@/models/stores/chainSwapConfig";
+import { useI18n } from 'vue-i18n';
+import { SwapUtils } from '@/util/swapUtils';
+import { Helper } from "@/util/typeHelper";
+
+
+export default {
+  name: "ViewServicesStableCoinsCheckStatus",
+
+  setup(){
+
+    const {t} = useI18n();
+    const route = useRoute();
+
+    const isLoaded = shallowRef(false);
+
+    let swapData = new ChainSwapConfig(networkState.chainNetworkName);
+    swapData.init();
+
+    const customErrorMessage = ref('');
+    const transactionHash = ref('');
+    const hashType = ref('BSC');
+
+    const disabledCheckStatus = computed(() => {
+      if(isLoaded.value){
+        return true;
+      }
+      if(hashType.value == 'BSC'){
+        // check for BSC txn type
+        if(transactionHash.value.length == 66){
+          if(transactionHash.value.substring(0, 2).toLocaleUpperCase() == '0X'){
+            return false
+          }
+          return true;
+        }else{
+          return true;
+        }
+      }else{
+        // check for sirius hash
+        if(transactionHash.value.length == 64){
+          return false;
+        }else{
+          return true;
+        }
+      }
+    });
+
+    const isSuccess = shallowRef(false);
+
+    const amount = ref('');
+    const siriusAddress = ref('');
+    const siriusTransactionHash = ref('');
+    const remoteTxnHash = ref('');
+    const txnStatus = ref('');
+    const time = ref('');
+    const explorerLink = () =>{
+    if(!networkState.currentNetworkProfile){
+        return ''
+    }
+    return networkState.currentNetworkProfile.chainExplorer.url + '/' + networkState.currentNetworkProfile.chainExplorer.hashRoute + '/';
+}
+    const remoteExplorerLink = ref('');
+    remoteExplorerLink.value = hashType.value === 'BSC'?swapData.BSCScanUrl:swapData.ETHScanUrl;
+
+    const checkStatus = async() => {
+      isLoaded.value = true;
+      isSuccess.value = false;
+      customErrorMessage.value = '';
+
+      try {
+        let url = '';
+        if(hashType.value == 'BSC'){
+          url = SwapUtils.getIncoming_BSCBuySiriusTokenCheckRemoteStatus_URL(swapData.swap_IN_SERVICE_URL, 'bsc', transactionHash.value);
+        }else{
+          url = SwapUtils.getIncoming_BSCBuySiriusTokenCheckStatus_URL(swapData.swap_IN_SERVICE_URL, transactionHash.value);
+        }
+        const response = await fetch(url, {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+
+        if(response.status == 200){ // data.status == 'fulfilled'
+        const json = await response.json();
+          remoteTxnHash.value = json.remoteTxnHash;
+          siriusTransactionHash.value = json.siriusTxnHash;
```

Comment:
siriusTxnHash might be undefined, when it is still pending (waiting for fund topup/ waiting next check interval).

Reply:
Will the siriusTxnHash value be returned as undefined if the txn is pending? Or the property will not be in the returned json?

Reply:
it will not return, property not exist yet